### PR TITLE
libboinc: fix include order

### PR DESCRIPTION
**Pull request by CyberTailor**
Compilation fails if "filesys.h" is included after "boinc_fcgi.h".

I don't know what's the reason but here's error message:

```
In file included from /usr/include/features.h:490,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from /usr/include/fcgi_stdio.h:18,
                 from boinc_fcgi.h:19,
                 from coproc.cpp:22:
/usr/include/wchar.h:582:24: error: 'malloc' attribute argument 1 is ambiguous
  582 |   __attribute_malloc__ __attr_dealloc_fclose;
      |                        ^~~~~~~~~~~~~~~~~~~~~
/usr/include/wchar.h:582:24: note: use a cast to the expected type to disambiguate
```

### CODE REVIEW

File: lib/coproc.cpp [hunk 1]
```diff
@@ -48,6 +43,12 @@
 #include "str_replace.h"
 #include "util.h"
 
+#ifdef _USING_FCGI_
```
Comment (AenBleidd) [inline lib/coproc.cpp:46]: Previously this piece of code was for non '_WIN32' only.
Could you please keep it?

**Review by AenBleidd — CHANGES_REQUESTED** [review]
Please specify os and compiler version where you see this error.

**Comment by CyberTailor** [comment]
> Please specify os and compiler version where you see this error.

Linux
GCC 12.2.1
Glibc 2.36

**Comment by AenBleidd** [comment]
There is at least one more place that is failing with the same issue: `lib/filesys.cpp`
Could you please fix that as well?
Ref:
- https://github.com/BOINC/boinc/actions/runs/4096879926/jobs/7064842121
- https://github.com/BOINC/boinc/actions/runs/4096879926/jobs/7064842000

And to test this fix please change this line https://github.com/BOINC/boinc/blob/aa31e3d2f8d7d8edfef1142f45d136dcd7fb870b/.github/workflows/linux.yml#L22
to `runs-on: ubuntu-latest`

**Comment by CyberTailor** [comment]
Missed that! Fixed now (hopefully)

**Comment by AenBleidd** [comment]
Still failing...

**Comment by AenBleidd** [comment]
I believe I found the root cause of this issue.
I'll try to fix it separately.
For now I'll keep this PR open in case I'm wrong

**Comment by AenBleidd** [comment]
Closing this in favor of #5096 that has a complete fix of the issue

**Pull request closed without merging**